src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1836,6 +1863,19 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $senderReferences[$serviceId] = new Reference($serviceId);
         }
 
+        $failureTransportsByTransportName = [];
+        foreach ($config['transports'] as $name => $transport) {
+            if ($transport['failure_transport']) {
+                if (!isset($config['transports'][$transport['failure_transport']])) {
+                    throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $transport['failure_transport']));
+                }
+
+                $failureTransportsByTransportName[$name] = $senderReferences[$transport['failure_transport']];
+            } else {
+                $failureTransportsByTransportName[$name] = $senderReferences[$config['failure_transport']] ?? null;
+            }
+        }
+
```

Comment:
Could this entire block be moved up to the first `foreach ($config['transports'] as $name => $transport) {`?

Reply:
Not really, because if you see I depend on `$senderReferences` which depend on `$senderAliases` being generated on that first  `foreach ($config['transports'] as $name => $transport)`,